modules/core/src-jvm-native/smithy4s/Timestamp.scala

```diff
@@ -39,6 +39,16 @@ case class Timestamp private (epochSecond: Long, nano: Int)
 
   def conciseDate: String = formatToString(2)
 
+  /**
+    * @return a copy of this timestamp with a milisecond resolution
+    */
+  def truncateNanos: Timestamp = copy(nano = (nano / 1000000) * 1000000)
```

Comment:
I'm finding the naming pretty confusing. I think it should be `truncateToSeconds` and `truncateToMillis` to make it clearer. For instance `truncateToSeconds` would truncate anything more precise than seconds.

Reply:
Alternatively : `toSecondPrecision` and `toMillsecondPrecision` ?

Reply:
renamed as `truncateToSeconds` etc.